src/AudioUnit/AudioComponent.cs

```diff
@@ -224,6 +377,58 @@ public XamCore.AppKit.NSImage GetIcon ()
 			return new XamCore.AppKit.NSImage (AudioComponentGetIcon (handle));
 		}
 #endif
+		[NoWatch, NoTV, Mac (10,13), iOS (11,0)]
+		[DllImport (Constants.AudioUnitLibrary, EntryPoint = "AudioUnitExtensionSetComponentList")]
+		static extern int /* OSStatus */ SetComponentList (IntPtr /* CFString */ extensionIdentifier, /* CFArrayRef */ IntPtr audioComponentInfo);
+
+		[NoWatch, NoTV, Mac (10,13), iOS (11,0)]
+		[DllImport (Constants.AudioUnitLibrary, EntryPoint = "AudioUnitExtensionCopyComponentList")]
+		static extern /* CFArrayRef */ IntPtr CopyComponentList (IntPtr /* CFString */ extensionIdentifier);
+
+		[NoWatch, NoTV, Mac (10,13), iOS (11,0)]
+		public AudioComponentInfo[] ComponentList {
+			get {
+				using (var cfString = new CFString (Name)) {
+					var cHandle = CopyComponentList (cfString.Handle);
+					if (cHandle == IntPtr.Zero)
+						return null;
+					using (var nsArray = Runtime.GetNSObject<NSArray> (cHandle)) {
```

Comment:
You own the returned CFArray, so use `GetNSObject<NSArray> (cHandle, owns: true)`.